jol-core/src/main/java/org/openjdk/jol/ljv/GraphvizVisualization.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.jol.ljv;
+
+import org.openjdk.jol.ljv.nodes.ArrayNode;
+import org.openjdk.jol.ljv.nodes.Node;
+import org.openjdk.jol.ljv.nodes.ObjectNode;
+
+import java.util.IdentityHashMap;
+
+public class GraphvizVisualization implements Visualization {
+    private final StringBuilder out = new StringBuilder();
+    private final LJV ljv;
+    private final IdentityHashMap<Object, String> alreadyDrawnObjectsIds = new IdentityHashMap<>();
+    private boolean alreadyDrawnNull = false;
+
+    public GraphvizVisualization(LJV ljv) {
+        this.ljv = ljv;
+    }
+
+    @Override
+    public boolean alreadyVisualized(Object obj) {
+        if (obj == null) {
+            return alreadyDrawnNull;
+        }
+
+        return alreadyDrawnObjectsIds.containsKey(obj);
+    }
+
+    @Override
+    public void diagramBegin() {
+        out.setLength(0); // Clearing String Builder before starting new DOT
+        out.append("digraph Java {\n")
```

Comment:
No it won't be a problem for Approvals tests, as far as I understand it can treat the line ending discrepancies correctly.
However I can notice `\n` in `OptionFormatter` and `ClassLayout`, so maybe postpone the line ending problem?

Reply:
Okay, sounds fine.